collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormDownloader.java

```diff
@@ -1,30 +1,538 @@
 package org.odk.collect.android.formmanagement;
 
+import android.net.Uri;
+
+import org.javarosa.core.reference.ReferenceManager;
 import org.odk.collect.android.R;
 import org.odk.collect.android.application.Collect;
-import org.odk.collect.android.utilities.MultiFormDownloader;
+import org.odk.collect.android.listeners.FormDownloaderListener;
+import org.odk.collect.android.provider.FormsProviderAPI;
+import org.odk.collect.android.utilities.FileUtils;
+import org.odk.collect.android.utilities.FormNameUtils;
+import org.odk.collect.android.utilities.Validator;
+import org.odk.collect.forms.Form;
+import org.odk.collect.forms.FormsRepository;
+import org.odk.collect.server.FormApiException;
+import org.odk.collect.server.FormListApi;
+import org.odk.collect.server.MediaFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
 
-import java.util.Collections;
-import java.util.HashMap;
+import timber.log.Timber;
+
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+import static org.odk.collect.utilities.PathUtils.getAbsoluteFilePath;
 
 /**
- * Provides a sarcophagus for {@link org.odk.collect.android.utilities.MultiFormDownloader} so it
+ * Provides a sarcophagus for {@link MultiFormDownloader} so it
  * can eventually be disposed of.
  */
 public class ServerFormDownloader implements FormDownloader {
 
     private final MultiFormDownloader multiFormDownloader;
```

Comment:
The `MultiFormDownloader` lives on in here as pulling it apart entirely is going to be pretty time-consuming and we've already hit the core goal of getting rid of any references to it in the rest of the code.